<details><summary>Failed to submit graph for 6fde578fbdcc7164b7df9d96cad28177bb5cbd5c
</summary>Message:
Failed to run command: 'pbp generate from-feedstock --output /tmp/tmpm48hj3hl/pydantic-extra-types-feedstock.yaml --custom-metadata-file /tmp/tmpm48hj3hl/custom_metadata.yaml --release-name pydantic-extra-types-feedstock-3-6fde578fbdcc7164b7df9d96cad28177bb5cbd5c pydantic-extra-types-feedstock/recipe', with error: 'INFO:root:Found feedstock pydantic-extra-types-feedstock at /tmp/tmpm48hj3hl/pydantic-extra-types-feedstock/recipe
INFO:root:Build/Test graph from pydantic-extra-types-feedstock
Created by sbx_user1051 on 2026-01-12 12:09:53
Command line: /opt/conda/envs/webhook-listener/bin/pbp generate from-feedstock --output /tmp/tmpm48hj3hl/pydantic-extra-types-feedstock.yaml --custom-metadata-file /tmp/tmpm48hj3hl/custom_metadata.yaml --release-name pydantic-extra-types-feedstock-3-6fde578fbdcc7164b7df9d96cad28177bb5cbd5c pydantic-extra-types-feedstock/recipe

INFO:root:Found abs.yaml in /tmp/tmpm48hj3hl/pydantic-extra-types-feedstock
INFO:pbpcli.utils.common:Setting env var: ANACONDA_ROCKET_ENABLE_PY314 = True.
INFO:root:Creating graph from feedstock at: pydantic-extra-types-feedstock
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.08 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 3 unchanging vars:(455), changing vars (3): ['blas_impl', 'numpy', 'python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
WARNING: Number of parsed outputs does not match detected raw metadata blocks. Identified output block may be wrong! If you are using Jinja conditionals to include or exclude outputs, consider using `skip: true  # [condition]` instead.
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
INFO:pbpcli.utils.common:Unsetting env var: ANACONDA_ROCKET_ENABLE_PY314
Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/bin/pbp", line 10, in <module>
    sys.exit(cli())
             ~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 413, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 624, in from_feedstock
    build_configs, test_configs = expand_build_matrix(feedstock.recipe_dir_path, src_channels)
                                  ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 79, in expand_build_matrix
    _targets, build_configs, test_configs = variants_for_feedstock_and_platform(
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        recipe_dir_path, platform, src_channels
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/utils/expand_variants.py", line 544, in variants_for_feedstock_and_platform
    metadata_tuples = api.render(
        parsed.recipe,
    ...<2 lines>...
        variants=None,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/api.py", line 68, in render
    return render_metadata_tuples(
        metadata_tuples,
    ...<3 lines>...
        bypass_env_check=bypass_env_check,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 1097, in render_metadata_tuples
    for od, om in meta.get_output_metadata_set(
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        permit_unsatisfiable_variants=permit_unsatisfiable_variants,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        permit_undefined_jinja=not finalize,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        bypass_env_check=bypass_env_check,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ):
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 2802, in get_output_metadata_set
    _check_run_constrained(output_tuples)
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 521, in _check_run_constrained
    raise RecipeError("\n".join(["", *errors]))
conda_build.exceptions.RecipeError: 
- Output 'pydantic-extra-types' has invalid run_constrained item: cron-converter >=1.2.2". Reason: Invalid spec 'cron-converter >=1.2.2"': Invalid version '1.2.2"': invalid character(s)
'
aws lambda request id: 60dda0ce-297f-53e2-9a57-c110cfb26223
[View in Grafana](https://grafana.anacondaconnect.com/explore?schemaVersion=1&panes=%7B%2286g%22:%7B%22datasource%22:%22loki-central%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7B__aws_cloudwatch_log_group%3D%5C%22%2Faws%2Flambda%2Fpbp-graph-submitter-prod%5C%22%7D%20%7C%20json%20%7C%20requestId%20%3D%20%6060dda0ce-297f-53e2-9a57-c110cfb26223%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-central%22%7D,%22editorMode%22:%22builder%22,%22direction%22:%22backward%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22logs%22:%7B%22columns%22:%7B%220%22:%22Time%22,%221%22:%22Line%22%7D,%22visualisationType%22:%22table%22,%22labelFieldName%22:%22labels%22%7D%7D%7D%7D&orgId=1)</details>

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -34,7 +34,7 @@ requirements:
     - pytz >=2024.1
     - python-tzdata >=2024.1
     - tzdata >=2024.1
-    - cron-converter >=1.2.2"
+    - cron-converter >=1.2.2
 
 test:
   source_files:
@@ -54,10 +54,9 @@ test:
   commands:
     - pip check
     # 1. missing python-ulid
-    # 2. test_json_schema is skipped because of AssertionError: assert {'properties'...pe': 'object'} == {'properties'...pe': 'object'}.
-    # 3. test_pendulum_dt is skipped because it fails on py314 due to pendulum 3.1.0 not supporting py314.
-    - pytest -v --ignore=tests/test_ulid.py --ignore=tests/test_json_schema.py --ignore=tests/test_pendulum_dt.py -k "not test_json_schema" tests  # [py==314]
-    - pytest -v --ignore=tests/test_ulid.py --ignore=tests/test_json_schema.py tests  # [py<314]
+    # 2. test_pendulum_dt is skipped because it fails on py314 due to pendulum 3.1.0 not supporting py314.
+    - pytest -v --ignore=tests/test_ulid.py --ignore=tests/test_pendulum_dt.py tests  # [py==314]
+    - pytest -v --ignore=tests/test_ulid.py tests  # [py<314]
 
 about:
   home: https://github.com/pydantic/pydantic-extra-types

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -54,9 +54,10 @@ test:
   commands:
     - pip check
     # 1. missing python-ulid
-    # 2. test_pendulum_dt is skipped because it fails on py314 due to pendulum 3.1.0 not supporting py314.
-    - pytest -v --ignore=tests/test_ulid.py --ignore=tests/test_pendulum_dt.py tests  # [py==314]
-    - pytest -v --ignore=tests/test_ulid.py tests  # [py<314]
+    # 2. test_json_schema and test_cron are skipped because cron-converter isn't available on the main channel.
+    - pytest -v --ignore=tests/test_ulid.py --ignore=tests/test_json_schema.py --ignore=tests/test_cron.py tests  # [py<314]
+    # test_pendulum_dt is skipped because it fails on py314 due to pendulum 3.1.0 not supporting py314.
+    - pytest -v --ignore=tests/test_ulid.py --ignore=tests/test_json_schema.py --ignore=tests/test_cron.py --ignore=tests/test_pendulum_dt.py tests  # [py==314]
 
 about:
   home: https://github.com/pydantic/pydantic-extra-types

